packages/zwave-js/src/lib/commandclass/BarrierOperatorCC.ts

```diff
@@ -367,6 +467,16 @@ export class BarrierOperatorCCEventReport extends BarrierOperatorCC {
 	public get state(): SignalState {
 		return this._state;
 	}
+
```

Comment:
Question: Does BarrierOperatorCCEventSet allow configuring a different signal state for every type? Then the implementation is not correct.

Reply:
every type has on and off state

2 types: Audible/Visual and Common states: On/Off
![Screenshot 2021-01-16 170024](https://user-images.githubusercontent.com/4048920/104815313-644c9080-581c-11eb-857c-cfa45771e498.jpg)

Reply:
So, in that case, state and type are not two separate properties. Instead you have a map where `type` is the key and `state` the value for that key. This also makes the SET_VALUE API more complicated.

An example would be the `Color Switch CC` (`targetColor` and `currentColor`).
==> https://github.com/zwave-js/node-zwave-js/blob/e1c251f5430f5637f1948d9443068d8c479e586f/packages/zwave-js/src/lib/commandclass/ColorSwitchCC.ts#L265-L287
and
==> https://github.com/zwave-js/node-zwave-js/blob/e1c251f5430f5637f1948d9443068d8c479e586f/packages/zwave-js/src/lib/commandclass/ColorSwitchCC.ts#L494-L500